### PR TITLE
VideoPress: Provide the list of allowed video extensions on the client initial state

### LINKED ISSUE #26468: VideoPress: expose allowed mime types of the site to the client
Probably, we can get the allowed mime types by using [get_allowed_mime_types()](https://developer.wordpress.org/reference/functions/get_allowed_mime_types/) core function and expose it in [the Initial state global var.](https://github.com/Automattic/jetpack/blob/a0c086638e0018e50d0264ec4312f276d7f3f111/projects/packages/videopress/src/class-admin-ui.php#L124)

### PR DESCRIPTION
**Pull request by lhkowalski**
Fixes #26468.

#### Changes proposed in this Pull Request:

* Use the list returned by `get_allowed_mime_types()` to generate a list of `video/*` file extensions
* Provide this list as a parameter on the client initial state data
* Each extension is associated with it's respective mime type, as per @retrofox sugestion [below](https://github.com/Automattic/jetpack/pull/26537#discussion_r984025177)

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

N/A.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Run `jetpack docker wp shell`
* On the new shell that opens, run `Automattic\Jetpack\VideoPress\Admin_UI::initial_state()`
* You should be able to see the new parameter, `allowedVideoExtensions`, on the list of parameters returned
* Example:

```
[
    ...
    "allowedVideoExtensions" => [
        "asf" => "video/x-ms-asf",
        "asx" => "video/x-ms-asf",
        "wmv" => "video/x-ms-wmv",
        "wmx" => "video/x-ms-wmx",
        "wm" => "video/x-ms-wm",
        "avi" => "video/avi",
        "divx" => "video/divx",
        "flv" => "video/x-flv",
        "mov" => "video/quicktime",
        "qt" => "video/quicktime",
        "mpeg" => "video/mpeg",
        "mpg" => "video/mpeg",
        "mpe" => "video/mpeg",
        "mp4" => "video/mp4",
        "m4v" => "video/mp4",
        "ogv" => "video/ogg",
        "webm" => "video/webm",
        "mkv" => "video/x-matroska",
        "3gp" => "video/3gpp",
        "3gpp" => "video/3gpp",
        "3g2" => "video/3gpp2",
        "3gp2" => "video/3gpp2",
        "videopress" => "video/videopress",
    ],
]"
```

### CODE REVIEW

File: projects/packages/videopress/src/class-admin-ui.php [hunk 1]
```diff
@@ -124,18 +143,19 @@ public static function render_initial_state() {
...
+			'paidFeatures'           => array(
 				'isVideoPressSupported'          => Current_Plan::supports( 'videopress' ),
 				'isVideoPress1TBSupported'       => Current_Plan::supports( 'videopress-1tb-storage' ),
 				'isVideoPressUnlimitedSupported' => Current_Plan::supports( 'videopress-unlimited-storage' ),
 			),
-			'productData'       => My_Jetpack_Products::get_product( 'videopress' ),
-			'siteSuffix'        => ( new Status() )->get_site_suffix(),
+			'productData'            => My_Jetpack_Products::get_product( 'videopress' ),
+			'siteSuffix'             => ( new Status() )->get_site_suffix(),
+			'allowedVideoExtensions' => self::get_allowed_video_extensions(),
```
Comment (retrofox) [inline projects/packages/videopress/src/class-admin-ui.php:162]: can we expose a `allowedVideoMimeTypes` or something like that too?

**Review by retrofox — COMMENTED** [review]
image

like a charm <3

File: projects/packages/videopress/src/class-admin-ui.php (hunk 1 above)
Comment (lhkowalski) [reply projects/packages/videopress/src/class-admin-ui.php:162]: I have two ideas on how to do it:

* having a separated list of `allowedVideoMimeTypes`, like this:

```
...
"allowedVideoMimeTypes" => [
    "video/x-ms-asf",
    "video/x-ms-wmv",
    ...
],
```

* returning it after each extension, sort of like the `get_allowed_mime_types()` functions does, like this:

```
...
"allowedVideoExtensions" => [
    "asf" => "video/x-ms-asf",
    "asx" => "video/x-ms-asf", // repeated mime type, just to be associated with each possible extension
    "wmv" => "video/x-ms-wmv",
    ...
],  
```

Which one do you think is better?

Comment (retrofox) [reply projects/packages/videopress/src/class-admin-ui.php:162]: second one. we are plenty of excellent native methods to process the data on the client side to get what we need.

Comment (lhkowalski) [reply projects/packages/videopress/src/class-admin-ui.php:162]: Done, now the returned value is as follow:

```
"allowedVideoExtensions" => [
    "asf" => "video/x-ms-asf",
    "asx" => "video/x-ms-asf",
    "wmv" => "video/x-ms-wmv",
    "wmx" => "video/x-ms-wmx",
    "wm" => "video/x-ms-wm",
    "avi" => "video/avi",
    "divx" => "video/divx",
    "flv" => "video/x-flv",
    "mov" => "video/quicktime",
    "qt" => "video/quicktime",
    "mpeg" => "video/mpeg",
    "mpg" => "video/mpeg",
    "mpe" => "video/mpeg",
    "mp4" => "video/mp4",
    "m4v" => "video/mp4",
    "ogv" => "video/ogg",
    "webm" => "video/webm",
    "mkv" => "video/x-matroska",
    "3gp" => "video/3gpp",
    "3gpp" => "video/3gpp",
    "3g2" => "video/3gpp2",
    "3gp2" => "video/3gpp2",
    "videopress" => "video/videopress",
],
```

Thank you for the suggestion!

**Review by retrofox — APPROVED** [review]
LGTM!

**Pull request merged**